### ISSUE TITLE
Multi-line arrow functions

### ISSUE THREAD

**Issue by ksmithbaylor** [pangloss/vim-javascript#332]
Currently, when the body of an arrow function that returns an immediate value is split over multiple lines, the second line (the value) is aligned even with the line above:

```javascript
const someVeryLongIdentifier = (a, b, c) =>
anotherVeryLongExpressionThatWillNotFitOnASingleLineNicely();

nextLine();
```

In my opinion, it should be indented one level more than the line on which it started, but still allow the following lines to stay at the same level as before:

```javascript
const someVeryLongIdentifier = (a, b, c) =>
  anotherVeryLongExpressionThatWillNotFitOnASingleLineNicely();

nextLine();
```

This should also apply to arrow functions defined as function arguments. Current behavior:

```javascript
someVeryLongIdentifier.getThings().filter(Boolean).map(x =>
                                                       anotherVeryLongExpressionThatWillNotFitOnASingleLineNicely());

                                                       nextLine();
```

Expected behavior:

```javascript
someVeryLongIdentifier.getThings().filter(Boolean).map(x =>
  anotherVeryLongExpressionThatWillNotFitOnASingleLineNicely());

nextLine();
```

**Comment by WChargin** [comment]
The following patch works for me. Can someone familiar with the repo make sure this seems reasonable? If it does, I'll pull-request it.

```diff
diff --git a/indent/javascript.vim b/indent/javascript.vim
index a935a87..f1329cb 100644
--- a/indent/javascript.vim
+++ b/indent/javascript.vim
@@ -55,7 +55,7 @@ let s:continuation_regex = '\%([\\*+/.:]\|\%(<%\)\@<![=-]\|\W[|&?]\|||\|&&\|[^=]
 " TODO: this needs to deal with if ...: and so on
 let s:msl_regex = s:continuation_regex.'|'.s:expr_case
 
-let s:one_line_scope_regex = '\<\%(if\|else\|for\|while\)\>[^{;]*' . s:line_term
+let s:one_line_scope_regex = '\%(\<\%(if\|else\|for\|while\)\>\|=>\)[^{;]*' . s:line_term
 
 " Regex that defines blocks.
 let s:block_regex = '\%([{[]\)\s*\%(|\%([*@]\=\h\w*,\=\s*\)\%(,\s*[*@]\=\h\w*\)*|\)\=' . s:line_term

```

I just added it to the one-liners regex. I had to put it in another non-capturing group so that the `=>` was separate from the `\>`, because `>` doesn't count as a word boundary.

**Comment by davidchambers** [comment]
This looks reasonable to me. Also, none of the current maintainers is familiar with the indentation-related code, so you may be the closest thing we have to an expert. ;)

**Comment by WChargin** [comment]
*Expert?* That's a scary thought! :)

I opened #339.

**Comment by davidchambers** [comment]
Resolved by #339
